src/mtcp/mtcp_restart.c

```diff
@@ -463,11 +463,25 @@ static void restart_slow_path()
   restorememoryareas(&rinfo);
 }
 
-static void mtcp_simulateread(int fd)
+static void mtcp_simulateread(int fd, MtcpHeader *mtcpHdr)
 {
   int mtcp_sys_errno;
+
+  // Print miscellaneous information:
+  char buf[MTCP_SIGNATURE_LEN+1];
+  mtcp_memcpy(buf, mtcpHdr->signature, MTCP_SIGNATURE_LEN);
+  buf[MTCP_SIGNATURE_LEN] = '\0';
+  mtcp_printf("\nMTCP: %s", buf);
+  mtcp_printf("**** original mtcp: %p-%p\n", mtcpHdr->restore_addr,
+              mtcpHdr->restore_addr + mtcpHdr->restore_size);
+  mtcp_printf("**** DMTCP entry point (ThreadList::postRestart()): %p\n",
+              mtcpHdr->post_restart);
+  mtcp_printf("**** brk (sbrk(0)): %p\n", mtcpHdr->saved_brk);
+  mtcp_printf("**** vdso: %p-%p\n", mtcpHdr->vdsoStart, mtcpHdr->vdsoEnd);
+  mtcp_printf("**** vvar: %p-%p\n", mtcpHdr->vvarStart, mtcpHdr->vvarEnd);
+
   Area area;
-  MTCP_PRINTF("Listing ckpt image area:\n");
+  mtcp_printf("\n**** Listing ckpt image area:\n");
```

Comment:
Same comment as before.  readelf.sh doesn't report the internal function and line number delivering the information to us.  util/readdmtcp.sh shouldn't be doing that either.